odbc-api/src/result_set_metadata.rs

```diff
@@ -4,9 +4,244 @@ use odbc_sys::SqlDataType;
 
 use crate::{
     handles::{slice_to_utf8, AsStatementRef, SqlChar, Statement},
-    ColumnDescription, DataType, Error,
+    sleep::{self, wait_for},
+    ColumnDescription, DataType, Error, Sleep,
 };
 
+/// Provides Metadata of the resulting the result set. Implemented by `Cursor` types and prepared
+/// queries. Fetching metadata from a prepared query might be expensive (driver dependent), so your
+/// application should fetch the Metadata it requires from the `Cursor` if possible.
+///
+/// See also:
+/// <https://docs.microsoft.com/en-us/sql/odbc/reference/develop-app/result-set-metadata>
+
+pub trait AsyncResultSetMetadata: AsStatementRef {
```

Comment:
This trait is basically 1:1 `ResultSetMetadata` trait with implementation tweaks to account for methods that are async

Reply:
So far I dodged thinking about async traits and their implications. Especially whether I can safely claim the resulting futures to be `Send`. I doged this issue so far by having `BlockCursorPolling` not implementing any `Cursor` trait, but just offering everything directly in the implementation. Yet I can also see Rust made some advances in async support since I last touched the issue.

Since there are many ways to implement asynchronous behavior in ODBC I would probably prefer the prefix `Polling` instead of `Async`.

Since this would require many tests I would also fine contributing and reviewing this trait in multiple steps, splitting among supported functions.